### PR TITLE
Material 3 updates page

### PR DESCRIPTION
**Pull request by MiraMarshall**
_Description of what this PR is changing or adding, and why:_
[Staging site](https://sm-material3-docs.web.app/development/ui/material3-guide)
This PR adds a page about Material 3. 

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/development/ui/material3-guide.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+components, and tools. 
+It enables developers and designers to build optimal user experiences 
+on Flutter and other platforms. 
+Material 3 provides updated features like dynamic color, elevation, 
+and typography.
+
+## Key updates
+Though the migration to Material 3 continues, 
+it introduces several major changes from older Material versions. 
+You can demo the Material 3 features with the Material 3 Flutter Showcase.
```
Comment (parlough) [inline src/development/ui/material3-guide.md:22]: Maybe link this to the showcase? https://flutter.github.io/samples/web/material_3_demo

File: src/development/ui/material3-guide.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+To enable Material 3, construct [`ThemeData()`][] with the 
+[`useMaterial3` property][] set to `true`.
+In using this property, you choose, or opt-in to use Material 3. 
+
+```dart
+void main() {
+  runApp(
+    MaterialApp(
+      title: 'Material 3 Property',
+      theme: ThemeData(),
```
Comment (parlough) [inline src/development/ui/material3-guide.md:67]: Looks like this should be the opening of the constructor:

```suggestion
      theme: ThemeData(
```

File: src/development/ui/material3-guide.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+* [`IconButton`][]: Introduces two new types of IconButton, 
+standard and contained.  
+
+The community tracks the migration of Material 3 to Flutter in the 
+[Bring Material 3 to Flutter][] GitHub issue.
+
+## How to use Material 3 in your Flutter app
+
+By default, all Material widgets use Material 2.
+To enable Material 3, construct [`ThemeData()`][] with the
```
Comment (parlough) [inline src/development/ui/material3-guide.md:58]: Maybe remove the `()`:

```suggestion
To enable Material 3, construct [`ThemeData`][] with the 
```

File: src/development/ui/material3-guide.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+[`IconButton`]: {{site.api}}/flutter/material/IconButton-class.html
+[`MaterialApp()`]: {{site.api}}/flutter/material/MaterialApp-class.html
+[Material 3 - Flutter]: https://m3.material.io/develop/flutter
+[Migrating a Flutter app to Material 3]: https://blog.codemagic.io/migrating-a-flutter-app-to-material-3/
+[Migrating to Material Design 3]: https://material.io/blog/blog/blog/migrating-material-3
+[`NavigationBar`]: {{site.api}}/flutter/material/NavigationBar-class.html
+[`NavigationDrawer`]: https://m3.material.io/components/navigation-drawer/overview 
+[`OutlinedButton`]:  {{site.api}}/flutter/material/OutlinedButton-class.html
+[`SegmentedButton`]: https://m3.material.io/components/segmented-buttons/overview
+[`ThemeData()`]: {{site.api}}/flutter/material/ThemeData-class.html
```
Comment (parlough) [inline src/development/ui/material3-guide.md:107]:
```suggestion
[`ThemeData`]: {{site.api}}/flutter/material/ThemeData-class.html
```

**Review by parlough — COMMENTED** [review]
Some small comments from a quick read-through.

**Review by sfshaza2 — APPROVED** [review]
lgtm

File: src/development/ui/material3-guide.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+views for buttons.
+
+#### Styled buttons 
+* [`FilledButton`][]: Indicates important button actions. 
+* [`FilledButton.tonal`][]: Combines outlined and filled buttons.
+
+### Updated widgets
+The following list outlines some key widgets that support 
+Material 3 design in Flutter. 
+* [`ElevatedButton`][]
```
[the remark this answers is not in the document]
Comment (MiraMarshall) [reply src/development/ui/material3-guide.md]: Thanks for the suggestion! I'll add that to #7977

**Review by sfshaza2 — APPROVED** [review]
lgtm (Yes, I have a very minor quibble or two, but nothing to hold this up!)

**Pull request merged**